logconsumer.go

```diff
@@ -0,0 +1,36 @@
+package testcontainers
+
+// StdoutLog is the log type for STDOUT
+const StdoutLog = "STDOUT"
+
+// StdinLog is the log type for STDIN
+const StdinLog = "STDIN"
+
+// BothLog is the log type for both
+// STDIN and STDOUT
+const BothLog = "BOTH"
+
+// Log represents a message that was created by a process
+// LogType is either "STDIN", "STDOUT", or "BOTH",
+// Content is the byte contents of the message itself
+type Log struct {
+	LogType string
+	Content []byte
+}
+
+// NewLog will allocate memory for a Log and return a
+// reference to that log, it will also ensure that the inputs
+// are valid for the log
+func NewLog(content []byte, logType string) *Log {
+	return &Log{
+		Content: content,
+		LogType: logType,
+	}
+}
+
+// LogConsumer represents any object that can
+// handle a Log, it is up to the LogConsumer instance
+// what to do with the log
+type LogConsumer interface {
+	Accept(string)
```

Comment:
this should accept a `Log`, rather than just a string.  Right now it just accepts the contents of the log.  But the `Log` object would allow us to view the type as well